setup.py

```diff
@@ -17,31 +17,63 @@
 from __future__ import division
 from __future__ import print_function
 
-from setuptools import setup, find_packages
-
-
-setup(
-    name='amazon.ion',
-    version='0.7.0',
-    description='A Python implementation of Amazon Ion.',
-    url='http://github.com/amzn/ion-python',
-    author='Amazon Ion Team',
-    author_email='ion-team@amazon.com',
-    license='Apache License 2.0',
-
-    packages=find_packages(exclude=['tests*']),
-    namespace_packages=['amazon'],
-
-    install_requires=[
-        'six',
-        'jsonconversion'
-    ],
-
-    setup_requires=[
-        'pytest-runner',
-    ],
-
-    tests_require=[
-        'pytest',
-    ],
-)
+from setuptools import setup, find_packages, Extension
+
+C_EXT = True
+
+
+def run_setup():
+    if C_EXT:
+        print('C extension is enabled!')
+        kw = dict(
+            ext_modules=[
+                Extension(
+                    'amazon.ion.ionc',
+                    sources=['amazon/ion/ioncmodule.c'],
+                    include_dirs=['amazon/ion/ion-c-build/include',
+                                  'amazon/ion/ion-c-build/include/ionc',
+                                  'amazon/ion/ion-c-build/include/decNumber'],
+                    libraries=['ionc', 'decNumber'],
+                    library_dirs=['amazon/ion/ion-c-build/lib'],
+                    extra_link_args=['-Wl,-rpath,%s' % '$ORIGIN/ion-c-build/lib',  # LINUX
+                                     '-Wl,-rpath,%s' % '@loader_path/ion-c-build/lib'  # MAC
+                                     ],
+                ),
+            ],
+        )
```

Comment:
On linux system,

After building wheels, it's called something like amazon.ion-0.7.95-cp37-cp37m-linux_x86_64.whl while pypi only supports manylinux (right now is linux) wheels. However one solution is just simply rename the wheel to something like amazon.ion-0.7.95-cp37-cp37m-manylinux2010_x86_64.whl...... I'm going to do more research about this.